recipes/zyre/all/conanfile.py

```diff
@@ -0,0 +1,74 @@
+from conans import ConanFile, CMake, tools
+from sys import platform
+import re
+import os
+
+
+class ZyreConan(ConanFile):
+    name = "zyre"
+    version = "2.0.1"
+    license = "MPL-2.0"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/zeromq/zyre"
+    description = "Local Area Clustering for Peer-to-Peer Applications."
+    topics = ("conan", "zyre", "czmq", "zmq", "zeromq",
+              "message-queue", "asynchronous")
+    exports_sources = ['CMakeLists.txt', 'Findzyre.cmake']
+    settings = "os", "compiler", "build_type", "arch"
+    requires = "zeromq/4.3.2", "czmq/4.2.0"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "drafts": [True, False],
+    }
+    default_options = {
+        "shared": True,
+        "fPIC": True,
+        "drafts": False,
+    }
+    generators = ["cmake"]
+    _source_subfolder = "source_subfolder"
```

Comment:
```suggestion
    _cmake = None
    _source_subfolder = "source_subfolder"
```
This adds caching to cmake. (part 1/2)